docs/index.rst

```diff
@@ -24,6 +24,38 @@ database using the `PostgreSQL Wire Protocol`_.
 databases in Java.
 
 
+*****
+About
+*****
+
+Overview
+========
+
+- This JDBC driver is needed in certain scenarios like the one outlined at
```

Comment:
Yes, I've added the corresponding example with 6a6912e66c. But, true, I will be happy to add a more crystal-clear sentence if @matriv could provide a corresponding statement?

Reply:
Well, the example still works on the older versions of Apache Flink, and points into the right direction why we need the "legacy" JDBC driver. This detail does not change, as we still need it.

Reply:
> It also is not clear there, why we should not use the PG JDBC driver.
> I will be happy to add a more crystal-clear sentence about the "why".

Coming from [^1][^2], I think a good explanation could be:

> When using the `postgres://` JDBC driver prefix, this will implicitly select the corresponding **PostgreSQL dialect implementation for Apache Flink**. However, this will employ a few behaviors that strictly expect a PostgreSQL server on the other end, so that some operations will croak on databases which only offer wire-compatibility with PostgreSQL.

What do you think about it?

[^1]: https://github.com/crate/cratedb-flink-jobs/issues/3
[^2]: https://pypi.org/project/sqlalchemy-postgresql-relaxed/

Reply:
I've just added 159010bb29, in order to improve this detail further. What do you think about it?

Reply:
Another example is the one from https://github.com/crate/crate/issues/11608#issuecomment-1403625495 but it may not be something to mention

Reply:
I've added it to 1684ea59, thanks!